lib/network/modules/components/Edge.js

```diff
@@ -677,16 +686,73 @@ class Edge {
         // Ignore the orientations.
         this.labelModule.pointToSelf = true;
         var x, y;
-        var radius = this.options.selfReferenceSize;
-        if (node1.shape.width > node1.shape.height) {
+
+        var radius = this.options.selfReference.size;
+        var percentage = 0.125;
+        
```

Comment:
If you change the `percentage` to angle you'll be able to simply assign the self reference angle to it and have the edge label positioned away from the node no matter what angle was chosen which, at least in my opinion, would be much better. Also faster and simpler code.

As a bonus this would work correctly for all angles including negative or angles in thousands etc.

PS: Quick searched revealed that `Edge._pointOnCircle` is used only from this method so you can safely remove the percentage to angle conversion from it.